### ISSUE TITLE
Focus outline clipped for data points at the boundary of Heatmap

### ISSUE THREAD

**Issue by franklinefrancis** [highcharts/highcharts#15414]
#### Expected behaviour
Focus outline shows up correctly for all data points of Heatmap

#### Actual behaviour
Focus outline shown during keyboard navigation in Heatmap is clipped for data points at the boundary

![Heatmap Focus Clipped](https://user-images.githubusercontent.com/1227574/112712457-baf9d900-8ef5-11eb-9f29-b5a02330f4d3.PNG)

#### Live demo with steps to reproduce
https://jsfiddle.net/franklinefrancis/upd0bv3w/

#### Product version
8.2.0

#### Affected browser(s)
Chrome, Firefox

**Comment by KacperMadej** [comment]
Thank you for reporting the bug.

Minimal demo: https://jsfiddle.net/BlackLabel/5q0wbyfh/

a11y - fyi: @oysteinmoseng

**Comment by oysteinmoseng** [comment]
Internal note: This happens because the focus border is drawn as a sibling to the point graphic in order to ensure that the positioning is correct, and all parent transformations are applied correctly. We should instead draw the focus border as an element outside the series SVG group, and calculate its positioning relative to the chart. Making sure this works for all series types will be a bit of work, as there are many positioning cases to consider.
